packages/kbn-securitysolution-grouping/src/hooks/use_get_group_selector.tsx

```diff
@@ -112,33 +116,46 @@ export const useGetGroupSelector = ({
 
   const onChange = useCallback(
     (groupSelection: string) => {
+      let newSelectedGroups: string[] = [];
+      let sendTelemetry = true;
       // Simulate a toggle behavior when maxGroupingLevels is 1
       if (maxGroupingLevels === 1) {
-        setSelectedGroups([groupSelection]);
+        newSelectedGroups = [groupSelection];
       } else {
+        // if the group is already selected, remove it
         if (selectedGroups.find((selected) => selected === groupSelection)) {
+          sendTelemetry = false;
           const groups = selectedGroups.filter((selectedGroup) => selectedGroup !== groupSelection);
           if (groups.length === 0) {
-            setSelectedGroups(['none']);
+            newSelectedGroups = ['none'];
           } else {
-            setSelectedGroups(groups);
+            newSelectedGroups = groups;
           }
-          return;
+        } else {
+          newSelectedGroups = isNoneGroup([groupSelection])
+            ? [groupSelection]
+            : [
+                ...selectedGroups.filter((selectedGroup) => selectedGroup !== 'none'),
+                groupSelection,
+              ];
         }
-
-        const newSelectedGroups = isNoneGroup([groupSelection])
-          ? [groupSelection]
-          : [...selectedGroups.filter((selectedGroup) => selectedGroup !== 'none'), groupSelection];
-        setSelectedGroups(newSelectedGroups);
       }
 
-      // built-in telemetry: UI-counter
-      tracker?.(
-        METRIC_TYPE.CLICK,
-        getTelemetryEvent.groupChanged({ groupingId, selected: groupSelection })
-      );
+      setSelectedGroups(newSelectedGroups);
 
-      onGroupChange?.({ tableId: groupingId, groupByField: groupSelection });
+      if (sendTelemetry) {
+        // built-in telemetry: UI-counter
+        tracker?.(
+          METRIC_TYPE.CLICK,
+          getTelemetryEvent.groupChanged({ groupingId, selected: groupSelection })
+        );
+      }
```

Comment:
This check allows sending telemetry only when a group is selected (this keeps the current telemetry collection behaviour while also triggering onChange for when a group is unselected.)